File: Backend/SafalYatra/carOperator/getOperatorProfile.php
<?php

include "../config/connection.php";
include "../utils/authHelper.php";


if (!isset($_POST['token'])) {
    echo json_encode(array(
        "success" => false,
        "message" => "Token is required"
    ));
    die();
}

$token = $_POST['token']; // Storing the token from POST request

$operatorId = getOperatorId($token);

if (!$operatorId) {
    echo json_encode(array(
        "success" => false,
        "message" => "Invalid token"
    ));
    die();
}


$sql = "SELECT operator_name, phone_number, email, location, registration_number, image_url, added_date FROM car_operators WHERE operator_id = $operatorId";
$result = mysqli_query($connect, $sql);

if (!$result) {
    echo json_encode(array(
        "success" => false,
        "message" => "Failed to fetch operators data",

    ));

    die();
}
$operator = mysqli_fetch_assoc($result);
echo json_encode(array(
    "success" => true,
    "message" => "Operators data fetched successfully",
    "operators" => $operator

));

File: Frontend/safal_yatra/lib/app/models/getOperatorProfile.dart
// To parse this JSON data, do
//
//     final operatorProfileResponse = operatorProfileResponseFromJson(jsonString);

import 'dart:convert';

OperatorProfileResponse operatorProfileResponseFromJson(String str) =>
    OperatorProfileResponse.fromJson(json.decode(str));

String operatorProfileResponseToJson(OperatorProfileResponse data) =>
    json.encode(data.toJson());

class OperatorProfileResponse {
  final bool? success;
  final String? message;
  final Operators? operators;

  OperatorProfileResponse({
    this.success,
    this.message,
    this.operators,
  });

  factory OperatorProfileResponse.fromJson(Map<String, dynamic> json) =>
      OperatorProfileResponse(
        success: json["success"],
        message: json["message"],
        operators: json["operators"] == null
            ? null
            : Operators.fromJson(json["operators"]),
      );

  Map<String, dynamic> toJson() => {
        "success": success,
        "message": message,
        "operators": operators?.toJson(),
      };
}

class Operators {
  final String? operatorName;
  final String? phoneNumber;
  final String? email;
  final String? location;
  final String? registrationNumber;
  final dynamic imageUrl;
  final DateTime? addedDate;

  Operators({
    this.operatorName,
    this.phoneNumber,
    this.email,
    this.location,
    this.registrationNumber,
    this.imageUrl,
    this.addedDate,
  });

  factory Operators.fromJson(Map<String, dynamic> json) => Operators(
        operatorName: json["operator_name"],
        phoneNumber: json["phone_number"],
        email: json["email"],
        location: json["location"],
        registrationNumber: json["registration_number"],
        imageUrl: json["image_url"],
        addedDate: json["added_date"] == null
            ? null
            : DateTime.parse(json["added_date"]),
      );

  Map<String, dynamic> toJson() => {
        "operator_name": operatorName,
        "phone_number": phoneNumber,
        "email": email,
        "location": location,
        "registration_number": registrationNumber,
        "image_url": imageUrl,
        "added_date": addedDate?.toIso8601String(),
      };
}
